### PR TITLE
[batch] use the gcr repo for the cloud sdk

### PR DESCRIPTION
**Pull request by danking**
Also update all uses of cloud sdk to same version.

Seems good to do anyway. I hope the newer version of gcloud includes a faster gsutil. I also hope that using gcr will improve the pull time a bit.

### CODE REVIEW

File: batch/build-batch-worker-image-startup.sh [hunk 1]
```diff
@@ -40,7 +40,7 @@ curl -fsSL "https://github.com/GoogleCloudPlatform/docker-credential-gcr/release
 docker-credential-gcr configure-docker
 
 docker pull ubuntu:18.04
-docker pull google/cloud-sdk:269.0.0-alpine
+docker pull gcr.io/google.com/cloudsdktool/cloud-sdk:310.0.0-alpine
```
Comment (jigold) [inline batch/build-batch-worker-image-startup.sh:43]: Let's make a new disk image and update the instance creation code.

**Review by jigold — CHANGES_REQUESTED** [review]

File: batch/build-batch-worker-image-startup.sh (hunk 1 above)
Comment (danking) [reply batch/build-batch-worker-image-startup.sh:43]: OK, good, I also want to fold in some stuff from isntance_pool.py in here (like the daemon.json) to try and speed up worker start up.

I have to work on hail team and ATGU meeting prep, but will return to this next week.

**Comment by danking** [comment]
I moved a couple things over into the image itself rather than on startup. Saves us a restart of docker and a restart of fluentd. I think that'll save ~20 seconds of worker startup.

**Review by jigold — CHANGES_REQUESTED** [review]
Something failed with the docker rules:

```
________________ Test.test_verify_no_access_to_metadata_server _________________

self = <test.test_batch.Test testMethod=test_verify_no_access_to_metadata_server>

    def test_verify_no_access_to_metadata_server(self):
        builder = self.client.create_batch()
        j = builder.create_job(os.environ['HAIL_CURL_IMAGE'],
                               ['curl', '-fsSL', 'metadata.google.internal', '--max-time', '10'])
        builder.submit()
        status = j.wait()
>       assert status['state'] == 'Failed', status
E       AssertionError: {'batch_id': 69, 'cost': '$0.0000', 'duration': 8664, 'exit_code': 0, ...}
E       assert 'Success' == 'Failed'
E         - Success
E         + Failed
```

**Comment by danking** [comment]
OK, this should be good.

**Review by jigold — APPROVED** [review]

**Pull request merged**